dataland-accounting-service/src/main/kotlin/org/dataland/datalandaccountingservice/services/CreditsManager.kt

```diff
@@ -0,0 +1,67 @@
+package org.dataland.datalandaccountingservice.services
+
+import org.dataland.datalandaccountingservice.model.TransactionDto
+import org.dataland.datalandaccountingservice.repositories.BilledRequestRepository
+import org.dataland.datalandaccountingservice.repositories.TransactionRepository
+import org.dataland.datalandbackend.openApiClient.api.CompanyDataControllerApi
+import org.dataland.datalandbackend.openApiClient.infrastructure.ClientException
+import org.dataland.datalandbackendutils.exceptions.ResourceNotFoundApiException
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.math.BigDecimal
+import java.math.RoundingMode
+import java.util.UUID
+
+/**
+ * Service class for managing requests concerning Dataland credits.
+ */
+@Service("CreditsManager")
+class CreditsManager(
+    private val transactionRepository: TransactionRepository,
+    private val billedRequestRepository: BilledRequestRepository,
+    private val companyDataControllerApi: CompanyDataControllerApi,
+) {
+    companion object {
+        const val HTTP_STATUS_NOT_FOUND = 404
+    }
+
+    private fun validateCompanyId(companyId: UUID) {
+        try {
+            companyDataControllerApi.isCompanyIdValid(companyId.toString())
+        } catch (ex: ClientException) {
+            if (ex.statusCode == HTTP_STATUS_NOT_FOUND) {
+                throw ResourceNotFoundApiException(
+                    summary = "Company ID not found.",
+                    message = "Dataland does not know the company ID $companyId.",
+                )
+            } else {
+                throw ex
+            }
+        }
+    }
+
+    /**
+     * Post a Dataland credits transaction.
+     * @param transactionDto contains all relevant information about the transaction
+     * @return the saved transaction where all IDs are of type String
+     */
+    @Transactional
+    fun postTransaction(transactionDto: TransactionDto<UUID>): TransactionDto<String> {
+        validateCompanyId(transactionDto.companyId)
+        return transactionRepository.save(transactionDto.toTransactionEntity()).toTransactionDtoString()
+    }
+
+    /**
+     * Calculate the current Dataland credits balance of the specified company.
+     * @param companyId the ID of the company whose balance is to be calculated
+     * @return the current Dataland credits balance of the specified company
+     */
+    @Transactional(readOnly = true)
+    fun getBalance(companyId: UUID): BigDecimal {
+        validateCompanyId(companyId)
+        return (
+            transactionRepository.getTotalBalanceFromTransactions(companyId) -
+                billedRequestRepository.getTotalCreditDebtFromBilledRequests(companyId)
+        ).setScale(1, RoundingMode.HALF_UP)
```

Comment:
This rounds all numbers to one digit after the dot so if somebody added 10.2456 credits in a post the balance shown will be 10.2, correct?